sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -109,9 +109,11 @@ export namespace ConnectionContext {
       await connectionContext.cbsSession.close();
 
       // Close the management sessions to ensure all the event handlers are released.
-      for (const id of Object.keys(connectionContext.clients)) {
-        const client = connectionContext.clients[id];
-        await (client as any)._context.managementClient.close();
+      for (const id of Object.keys(connectionContext.clientContexts)) {
+        const clientContext = connectionContext.clientContexts[id];
+        if (clientContext.managementClient) {
```

Comment:
Is it possible that you will close the managementClient while it is still in use by another client?

Reply:
This code path is reached when we get a `disconnected` event from the AMQP connection. All the `close()` in here are clean up operations i.e. everything must be closed. Therefore, it isnt required to check if a management client is shared between multiple clients or not